backend/service/audit/storage/sql/sql.go

```diff
@@ -187,18 +187,46 @@ func (c *client) query(ctx context.Context, query string, args ...interface{}) (
 	return events, nil
 }
 
+// We create our own connection to use for acquiring the advisory lock.
+// For advisory locks you must use the same session to issue the unlock
+// If not the lock will stay locked until that connection is severed or
+// the unlock request uses the same session in which the lock was created.
+func (c *client) getAdvisoryConn() (*sql.Conn, error) {
+	if c.advisoryLockConn != nil {
+		return c.advisoryLockConn, nil
+	}
+
+	advisoryLockConn, err := c.db.Conn(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	c.advisoryLockConn = advisoryLockConn
+	return advisoryLockConn, nil
+}
+
 func (c *client) AttemptLock(ctx context.Context, lockID uint32) (bool, error) {
+	conn, err := c.getAdvisoryConn()
+	if err != nil {
+		return false, err
+	}
+
 	var lock bool
-	if err := c.db.QueryRowContext(ctx, "SELECT pg_try_advisory_lock($1);", lockID).Scan(&lock); err != nil {
+	if err := conn.QueryRowContext(ctx, "SELECT pg_try_advisory_lock($1);", lockID).Scan(&lock); err != nil {
 		c.logger.Error("Unable to query for a advisory lock", zap.Error(err))
 		return false, err
 	}
 	return lock, nil
 }
 
 func (c *client) ReleaseLock(ctx context.Context, lockID uint32) (bool, error) {
+	conn, err := c.getAdvisoryConn()
+	if err != nil {
+		return false, err
+	}
+
 	var unlock bool
-	if err := c.db.QueryRowContext(ctx, "SELECT pg_advisory_unlock($1)", lockID).Scan(&unlock); err != nil {
+	if err := conn.QueryRowContext(ctx, "SELECT pg_advisory_unlock($1)", lockID).Scan(&unlock); err != nil {
```

Comment:
one optimization would be to invoke Close() after unlocking so the connection is returned to the pool so each worker is not always holding one connection constantly for this.